library/Fission/Prelude.hs

```diff
@@ -29,5 +30,8 @@ import RIO.List                    (headMaybe, intercalate)
 import RIO.Process
 import RIO.Time
 
+andThen :: (m a -> b) -> m a -> m b
+andThen = (<<=)
```

Comment:
Yeah, this isn't Elm. If anything, it should be called `bind`, but we may just want to use `>>=`. The term `andThen` has a lot of overlap with a BUNCH of other concepts, and bind doesn't _necessarily_ create asequence (it does often, but not always).

Reply:
@expede Gotcha 👍  So far example, you would do...

```
a
  |> b
  >>= c
  |> d
```

or:
1. Would you rather use `|> bind` in this scenario?
2. Or avoid this scenario all together and use a `do`, `let` or `where` block?

Reply:
In the specific case of the comment on `checkUser`, using a `do` block is clearer. `|>` and ``>>=` operate at different levels, so in the case that you _must_ pipeline, I'd use a `bind`, but also remember that you're likely going to get stuck inside that closure if you need to reuse that variable binding.

For science, the other other way to do it is to use a straight-up `fmap` instead of `bind` or `>>=`:

```haskell
checkUser
  :: MonadDatabase
  => BasicAuthData
  -> RIO cfg (BasicAuthResult User)
checkUser (BasicAuthData authUsername authPassword) =
  from (\user -> do
    where_ (byUsername user)
    return user)
  |> selectFirst
  |> fmap (maybe NoSuchUser checkPassword')

checkPassword' :: User -> BasicAuthResult User
checkPassword' usr =
  if validatePassword (encodeUtf8 <| secretDigest <| usr) password
     then Authorized usr
     else Unauthorized
```

Reply:
That looks clean af 👌 But yeah, I will stick with the previously suggested `selectFirst <| ...` format. Thanks for the science experiment 😄